From bfe9c573a9e376323929c80b2b71c59727fab0cc Mon Sep 17 00:00:00 2001
From: Chet Ramey <chet.ramey@case.edu>
Date: Fri, 17 Nov 2023 16:06:49 -0500
Subject: [PATCH 7/7] Readline-8.2 patch 7: display newline if readline is
 called without a prompt string and return is typed on a single empty line

---
 display.c  | 4 ++--
 patchlevel | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/display.c b/display.c
index ad573f9..803d4ce 100644
--- a/display.c
+++ b/display.c
@@ -3338,9 +3338,9 @@ _rl_update_final (void)
       puts_face (&last_line[_rl_screenwidth - 1 + woff],
 		 &last_face[_rl_screenwidth - 1 + woff], 1);
     }
-  _rl_vis_botlin = 0;
-  if (botline_length > 0 || _rl_last_c_pos > 0)
+  if ((_rl_vis_botlin == 0 && botline_length == 0) || botline_length > 0 || _rl_last_c_pos > 0)
     rl_crlf ();
+  _rl_vis_botlin = 0;
   fflush (rl_outstream);
   rl_display_fixed++;
 }
diff --git a/patchlevel b/patchlevel
index 6ebd6a5..e340b58 100644
--- a/patchlevel
+++ b/patchlevel
@@ -1,3 +1,3 @@
 # Do not edit -- exists only for use by patch
 
-6
+7
-- 
2.38.0.windows.1

